features/step_definitions/booking_steps.rb

```diff
@@ -0,0 +1,27 @@
+Given(/^booking data exists in the data warehouse$/) do
+  @begin_date = Date.parse('1/1/2016')
+  @end_date = Date.parse('3/1/2016')
+
+  PopulateDateDimension.new(begin_date: @begin_date, end_date: @end_date).call
+
+  @created = @begin_date.upto(@end_date)
+                        .map { |date| { date: date, reference_number: SecureRandom.uuid } }
+                        .map(&PopulateBookingFact.method(:new))
+                        .map(&:call)
+                        .size
+end
```

Comment:
I felt like the functional style makes it clear the transformations that are being carried out.